src/Compilers/CSharp/Portable/BoundTree/BoundDagEvaluation.cs

```diff
@@ -26,9 +33,10 @@ private Symbol Symbol
                 }
             }
         }
+
         public override int GetHashCode()
         {
-            return Hash.Combine(GetOriginalInput().GetHashCode(), this.Symbol?.GetHashCode() ?? 0);
```

Comment:
>this.Symbol?.GetHashCode() ?? 0) [](start = 66, length = 32)

Is this change needed?

Reply:
Yes.  We need to treat an evaluation of `X<object>.Field` as the same field as `X<dynamic>.Field`.

---
In reply to: [362567115](https://github.com/dotnet/roslyn/pull/40677#discussion_r362567115) [](ancestors = 362567115)

Reply:
I believe we ignore differences in `object` and `dynamic` in `GetHashCode()`  (see `DynamicTypeSymbol.GetHashCode()` for instance) so the hash codes for `X<object>.Field` and `X<dynamic>.Field` should be the same. 

And, it looks like `SymbolEqualityComparer.GetHashCode(ISymbol obj)` simply calls `obj.UnderlyingSymbol.GetHashCode()` as the previous code here was doing.

---
In reply to: [362689732](https://github.com/dotnet/roslyn/pull/40677#discussion_r362689732) [](ancestors = 362689732,362567115)